packages/oidc_core/lib/src/managers/user_manager_base.dart

```diff
@@ -642,7 +643,10 @@ abstract class OidcUserManagerBase {
 
     final idTokenNonce = newUser
         .parsedIdToken.claims[OidcConstants_AuthParameters.nonce] as String?;
-    if (nonce != null && idTokenNonce != nonce) {
+    final bytes = utf8.encode(rawNonce ?? '');
+    final hashedNonce = sha256.convert(bytes).toString();
+
+    if (idTokenNonce != hashedNonce) {
       logAndThrow(
         'Server returned a wrong id_token nonce, might be a replay attack.',
       );
```

Comment:
```suggestion
   if (rawNonce != null) {
    final bytes = utf8.encode(rawNonce);
    final hashedNonce = sha256.convert(bytes).toString(); 
    if (idTokenNonce != hashedNonce) {
      logAndThrow(
        'Server returned a wrong id_token nonce, might be a replay attack.',
      );   
    }
```

Reply:
I've made the change breaking because "nonce" inside the token now differs from"rawNonce". If someone for some federation reason implemented an equality check it would fail. This would be really difficult to discover. I'd rather have my code fail if I was using "getNonce()".

What do you think?

Reply:
Since we are still storing the raw nonce, the only breaking change is what's being sent in the request.
And what's sent in the request is already an implementation detail abstracted from the user (both nonce and state).

So what the user would see is just a different "look" of the nonce parameter, even if it has a special meaning in the package.